receivers/discord/config.go

```diff
@@ -16,12 +17,13 @@ type Config struct {
 	UseDiscordUsername bool   `json:"use_discord_username,omitempty" yaml:"use_discord_username,omitempty"`
 }
 
-func NewConfig(jsonData json.RawMessage) (Config, error) {
+func NewConfig(jsonData json.RawMessage, decryptFn receivers.DecryptFunc) (Config, error) {
 	var settings Config
 	err := json.Unmarshal(jsonData, &settings)
 	if err != nil {
 		return Config{}, fmt.Errorf("failed to unmarshal settings: %w", err)
 	}
+	settings.WebhookURL = decryptFn("url", settings.WebhookURL)
```

Comment:
Is this plain text at present? Does this mean existing Discord configurations will be broken and need configuration?

Reply:
Yes, it's plain text. The current configurations will not be broken because how `decryptFn` is implemented: if encrypted key is missing, it falls back to the plain text

Reply:
I see this doesn't break config sets for the old format, but what about reads? Will this field no longer be returned from API since it's now secure? If so, this could break how Terraform updates state.

Reply:
The field is returned as insecure until the next write that will convert it to a secure string, and provisioning API will start responding with `[REDACTED]`. 
I checked the Terraform provider and after the update, it detects the field as requiring change every time. So, this needs to be fixed in the terraform provider.

Reply:
This PR does not break anything, though unless there is another change in Grafana that promotes this field as secured.

Reply:
> it detects the field as requiring change every time

This will probably break the CI on https://github.com/grafana/terraform-provider-grafana/, if it runs against `main`.

The tests have a step that validates that you get a clean apply every time. Otherwise, the tests will start failing at the next Grafana release - we should really try to fix this in the provider before this change is released.

Reply:
Specifically, without this, lots of customer CIs will start behaving oddly, as many tools surrounding terraform (like Atlantis) rely on its idempotency and ability to detect when no changes will need to be made. I think with this break, Terraform will _always_ think a change needs to be made until a new version of the provider is released.

I think we should file an issue in the provider to track it.

Reply:
This particular change does not break anything. It just let the configuration read from two sources. The writing logic is in Grafana and is not touched.

Reply:
I'm trying to rectify:
> This particular change does not break anything. It just let the configuration read from two sources.

with:
> The field is returned as insecure until the next write that will convert it to a secure string, and provisioning API will start responding with [REDACTED].
I checked the Terraform provider and after the update, it detects the field as requiring change every time.

These statements seem contradictory to me, perhaps there is something I'm missing. If we're changing how a value is returned, and TF can't see it anymore and behaves incorrectly, that's a break. Clean applies are a requirement of Terraform - TF will no longer generate clean applies once the contact point is changed. Are we saying this PR doesn't cause this? Are you making multiple PRs as part of this change, is there another future PR that does cause this break, and we should fix Terraform to read both fields before that?